mono/metadata/icall-eventpipe.c

```diff
@@ -230,10 +230,24 @@ ves_icall_System_Diagnostics_Tracing_EventPipeInternal_Enable (
 
 	char *output_file_utf8 = mono_utf16_to_utf8 (output_file, g_utf16_len (output_file), error);
 
+	EventPipeProviderConfigurationNative *native_config_providers = (EventPipeProviderConfigurationNative *)providers;
+	EventPipeProviderConfiguration *config_providers = g_new0 (EventPipeProviderConfiguration, providers_len);
+
+	if (config_providers) {
```

Comment:
Is it intended that 'fix mono C++ build' also makes behavioral changes here? The changes look fine but are unexpected

Reply:
@lateralusX

Reply:
Jupp, there was an issue with existing since it used two incompatible types that only showed up in C++ build, so this needs to be fixed in order to build. That eventpipe code is also still work in progress (I'm working on it) and only used in dotnet/runtime, but this file is still synced (don't know if we can exclude individual files), but is protecred by #ifdef ENABLE_PERFTRACING so will only be build in dotnet/runtime.